### PR TITLE
[Core] Add an environment variable which needs to be set explicitly to allow BlockSpaceManagerV1

### PR DESCRIPTION
**Pull request by sroy745**
This PR introduces an environment variable that must be explicitly set to enable BlockSpaceManagerV1. This requirement informs users who have set use_v2_block_manager to False that BlockSpaceManagerV1 is being deprecated in favor of BlockSpaceManagerV2. As a result, users can test and transition to BlockSpaceManagerV2 in advance.

### CODE REVIEW

**Comment by sroy745** [comment]
The lora-test failures seem unrelated to my change and so is the speculative-decoding-tests  failures. The speculative-decoding-tests  failures I think should get fixed in https://github.com/vllm-project/vllm/pull/9160

@comaniac PTAL when you get a chance.

**Comment by DarkLight1337** [comment]
The CI failures will be fixed by #9160

**Comment by comaniac** [comment]
I unblocked rest tests manually to make sure we cover all cases. Let's see if it still passed

**Comment by comaniac** [comment]
Custom model test failed at main so can be ignored.

File: tests/basic_correctness/test_chunked_prefill.py [hunk 1]
```diff
@@ -20,6 +22,16 @@
...
 
 
+@pytest.fixture(scope="module", autouse=False)
+def check_deprecated_block_manager():
+    assert envs.VLLM_ALLOW_DEPRECATED_BLOCK_MANAGER_V1 is True, (
+        "To allow the use of deprecated BlockSpaceManagerV1, set the "
+        "environment variable VLLM_ALLOW_DEPRECATED_BLOCK_MANAGER_V1=1. "
+        "You can run the tests with: "
+        "`VLLM_ALLOW_DEPRECATED_BLOCK_MANAGER_V1=1 pytest tests/core/test_scheduler.py`"  #noqa
+    )
```
Comment (comaniac) [inline tests/basic_correctness/test_chunked_prefill.py:32]: Can we put this in conftest or utils so that you don't need to duplicate it multiple times?

Comment (sroy745) [reply tests/basic_correctness/test_chunked_prefill.py]: I moved it to utils and reusing it in all tests.

**Review by comaniac — APPROVED** [review]

**Pull request merged**